server/persistence/jdbc/src/main/java/JdbcStorageManager.java

```diff
@@ -0,0 +1,118 @@
+import io.whitefox.core.*;
+import io.whitefox.persistence.StorageManager;
+import io.whitefox.persistence.exceptions.InvalidPageTokenException;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+import mapper.ShareMapper;
+import repository.ShareRepository;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Transactional
+@ApplicationScoped
+public class JdbcStorageManager implements StorageManager {
+
+    @Inject
+    ShareRepository shareRepository;
+
+    @Inject
+    public JdbcStorageManager(){
+        this.shareRepository = new ShareRepository();
+    }
+
+    @Override
+    public Optional<Share> getShare(String share) {
+        return Optional.empty();
+    }
+
+    @Override
+    public ResultAndTotalSize<List<Share>> getShares(int offset, int maxResultSize) {
+
+        List<Share> shares = shareRepository.listAll().stream()
+                .map(ShareMapper::shareDaoToShare)
+                .collect(Collectors.toList());
+        var totalSize = shares.size();
+        if (offset > totalSize) {
+            throw new InvalidPageTokenException(
+                    String.format("Invalid Next Page Token: token %s is larger than totalSize", offset));
+        } else {
+            return new ResultAndTotalSize<>(
+                    shares.stream().skip(offset).limit(maxResultSize).collect(Collectors.toList()),
+                    totalSize);
+        }
```

Comment:
This eagerly fetches all shares, it might get very slow when a lot of shares are created.
Have a look at: https://quarkus.io/guides/hibernate-orm-panache#paging